scripts/benchmarks_local.py

```diff
@@ -0,0 +1,451 @@
+            repo = Repo(repo_path)
+            repo.git.checkout(commit)
+            repo.git.show('HEAD')
+        else:
+            repo = Repo(repo_path)
+            repo.remotes.origin.fetch()
+            repo.git.checkout(commit)
+            repo.git.show('HEAD')
+
+    # Determine what we need to generate for the local benchmarks
+    generate_all_runtype_dependencies(parsed_args, repo_path, commit, (is_local and not parsed_args.skip_local_rebuild) or parsed_args.rebuild_artifacts)
+
+# Run tests on the local machine
+def run_benchmarks(parsed_args: Namespace, commits: list) -> None:
+    # Generate the correct benchmarks_ci.py arguments for the run type
+    for run_type in enum_name_list_to_enum_list(RunType, parsed_args.run_type_names):
+        # Run the benchmarks_ci.py test and save results
+        try:
+            benchmark_ci_args = generate_benchmark_ci_args(parsed_args, run_type, commits)
+            getLogger().info(f"Running benchmarks_ci.py for {run_type} at {commits} with arguments \"{' '.join(benchmark_ci_args)}\".")
+            kill_dotnet_processes(parsed_args)
+            benchmarks_ci.__main(benchmark_ci_args) # Build the runtime includes a download of dotnet at this location
+        except CalledProcessError:
+            getLogger().error('benchmarks_ci exited with non zero exit code, please check the log and report benchmark failure')
+            raise
+
+        getLogger().info(f"Finished running benchmark for {run_type} at {commits}.")
+
+# Check if the specified references exist in the given repository URL.
+# If a reference does not exist, raise an exception.
+# 
+# Arguments:
+# - repo_url (str): The URL of the repository to check.
+# - references (list): A list of references (branches or commit hashes) to check.
+# - repo_storage_path (str): The directory where the cloned repository is stored.
+# - repo_dir (str): The name of the directory where the cloned repository is stored.
+#
+# Returns: None
+def check_references_exist_and_add_branch_commits(repo_url: str, references: list, repo_storage_path: str, repo_dir: str):
+    getLogger().debug(f"Inside check_references_exist_and_add_branch_commits: Checking if references {references} exist in {repo_url}.")
+    
+    # Initialize a new Git repository in the specified directory
+    repo_combined_path = os.path.join(repo_storage_path, repo_dir)
+    if not os.path.exists(repo_combined_path):
+        getLogger().debug(f"Cloning {repo_url} to {repo_combined_path}.")
+        repo = Repo.clone_from(repo_url, repo_combined_path)
+    else:
+        repo = Repo(repo_combined_path)
+        repo.remotes.origin.fetch()
+
+    # Check if each reference exists in the repository
+    for reference in references:
+        try:
+            result = repo.git.branch('-r', '--contains', reference) # Use git branch -r --contains <commit> to check if a commit is in a branch
+            if "error: malformed object" in result:
```

Comment:
What are the two different types of error that we are trying to handle here? I see we raise the same exception, is it just the case we can get in this state from two different paths?

Reply:
The "error: malformed object" is a leftover from when I was first figuring out the check for references. It will be removed in the next feedback commit.